test/test_h_transport_problem.py

```diff
@@ -1,11 +1,100 @@
 import festim as F
 import tqdm.autonotebook
 import mpi4py.MPI as MPI
-import dolfinx
+import dolfinx.mesh
+from dolfinx import fem, nls
 import ufl
 import numpy as np
+import pytest
+
+test_mesh = F.Mesh1D(vertices=np.array([0.0, 1.0, 2.0, 3.0, 4.0]))
+x = ufl.SpatialCoordinate(test_mesh.mesh)
+dummy_mat = F.Material(D_0=1, E_D=1, name="dummy_mat")
+
 
 # TODO test all the methods in the class
+@pytest.mark.parametrize(
+    "value", [1, fem.Constant(test_mesh.mesh, 1.0), 1.0, "coucou", lambda x: 2 * x[0]]
+)
+def test_temperature_setter_type(value):
+    """Test that the temperature type is correctly set"""
+    my_model = F.HydrogenTransportProblem(
+        mesh=test_mesh,
+    )
+
+    if not isinstance(value, (fem.Constant, int, float)):
+        if callable(value):
+            my_model.temperature = value
+        else:
+            with pytest.raises(TypeError):
+                my_model.temperature = value
+
+
+@pytest.mark.parametrize(
+    "input, expected_value",
+    [
+        (1.0, False),
+        (1, False),
+        (lambda t: t, True),
+        (lambda t: 1.0 + t, True),
+        (lambda x: 1.0 + x[0], False),
+        (lambda x, t: 1.0 + x[0] + t, True),
+        (lambda x, t: ufl.conditional(ufl.lt(t, 1.0), 100.0 + x[0], 0.0), True),
+    ],
+)
+def test_time_dependent_temperature_attribute(input, expected_value):
+    """Test that the temperature_time_dependent attribute is correctly set"""
+
+    my_model = F.HydrogenTransportProblem()
+    my_model.temperature = input
+
+    assert my_model.temperature_time_dependent == expected_value
+
+
+def test_define_temperature_value_error_raised():
+    """Test that a ValueError is rasied when the temperature is None"""
```

Comment:
```suggestion
    """Test that a ValueError is raised when the temperature is None"""
```

One day we'll get them all